### PR TITLE
Refactor editor mapitems

### PR DESCRIPTION
**Pull request by Marmare314**
Creating a separate PR for every change will be a bit tedious, so I split the changes by commits instead. I hope this is fine. Also I'm not sure how many commits can reasonably be included in a PR.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/mapitems/map.cpp [hunk 1]
```diff
@@ -177,3 +177,53 @@ void CEditorMap::MakeTuneLayer(const std::shared_ptr<CLayer> &pLayer)
...
+					++LayerIndex;
+				}
+				++GroupIndex;
+			}
+		}
+		++ImageIndex;
+	}
+}
+
+bool CEditorMap::HandleMapDrop(const char *pFileName, int StorageType)
```
Comment (Robyt3) [inline src/game/editor/mapitems/map.cpp:218]: This function accesses `m_pEditor` on almost every line so it should be a member function of `CEditor` directly.

File: src/game/editor/mapitems/map.cpp [hunk 2]
```diff
@@ -177,3 +177,53 @@ void CEditorMap::MakeTuneLayer(const std::shared_ptr<CLayer> &pLayer)
 	m_pTuneLayer = std::static_pointer_cast<CLayerTune>(pLayer);
 	m_pTuneLayer->m_pEditor = m_pEditor;
 }
+
+void CEditorMap::PerformSanityChecks(const std::function<void(const char *pErrorMessage)> &ErrorHandler)
```
Comment (Robyt3) [inline src/game/editor/mapitems/map.cpp:183]: This function is only used when loading maps so I'd say it also belongs in `map_io.cpp`.

**Comment by Robyt3** [comment]
Wouldn't it be easier if the editor components still had a constructor taking a `CEditor *`, so you don't have to call `Init` explicitly for every component?

File: src/engine/graphics.h [hunk 3]
```diff
@@ -76,19 +76,19 @@ class CImageInfo
...
 
 	/* Variable: height
 		Contains the height of the image */
-	int m_Height;
+	int m_Height = 0;
 
 	/* Variable: format
 		Contains the format of the image. See <Image Formats> for more information. */
-	int m_Format;
+	int m_Format = 0;
```
Comment (Robyt3) [inline src/engine/graphics.h:87]: Should use a `FORMAT_*` enum value.

File: src/game/editor/mapitems/map.cpp (hunk 1 above)
Comment (Marmare314) [reply src/game/editor/mapitems/map.cpp:218]: I asked about that on the PR that added this, but only after it was already merged.

File: src/game/editor/mapitems/sound.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#define GAME_EDITOR_MAPITEMS_SOUND_H
+
+#include <base/system.h>
+
+#include <game/editor/component.h>
+
+class CEditorSound : public CEditorComponent
+{
+public:
+	explicit CEditorSound(CEditor *pEditor);
```
Comment (Robyt3) [inline src/game/editor/mapitems/sound.h:11]: What's the reason only this is `explicit` but not the other constructors?

Comment (Marmare314) [reply src/game/editor/mapitems/sound.h:11]: No, forgot to add it to the others. But explicit is better here right?

Comment (Robyt3) [reply src/game/editor/mapitems/sound.h:11]: Yeah, seems like cleaner code to make conversions explicit instead of allowing them to be implicit.

**Review by Robyt3 — APPROVED** [review]

**Pull request merged**